### ISSUE TITLE
Bug: Do not add `/Zi` in `CMAKE_CXX_FLAGS_RELEASE`

### ISSUE THREAD

**Issue by mwestphal** [assimp/assimp#4845]
**Describe the bug**
Please do not add `/Zi` in the CMAKE_CXX_FLAGS_RELEASE. Why forces to create debug information when a developer may not need them.

I had to remove the option because of a sccache bug with cmake + ninja + .pdb files:
https://gitlab.kitware.com/cmake/cmake/-/issues/20222#note_1169887 

**To Reproduce**
1. Build assimp with sccache, msvc, ninja
2.
 
```
D:\a\f3d-superbuild\f3d-superbuild\build\superbuild\assimp\src\code\Common\Compression.cpp: fatal error C1041: cannot open program database 'D:\a\f3d-superbuild\f3d-superbuild\build\superbuild\assimp\build\code\CMakeFiles\assimp.dir\vc140.pdb'; if multiple CL.EXE write to the same .PDB file, please use /FS
```

**Expected behavior**
No build failure and no .pdb file

**Platform (please complete the following information):**
 - OS: Windows

**Comment by kimkulling** [comment]
On its way!

**Comment by kimkulling** [comment]
Fixed!

**Comment by mwestphal** [comment]
Thanks! I just removed our patch!

**Comment by RichardTea** [comment]
This change was incorrect and must be reverted, as it makes assimp impossible to support.
Symbols (PDBs) are absolutely necessary for debugging.

The ccache project currently recommends using `/Z7` when ccache is in use.
Apache Arrow uses an additional cmake config option to automatically swap out `/Zi` for `/Z7`.

**Comment by tellypresence** [comment]
Build issues tracked via epic #6189

**Comment by RichardTea** [comment]
The OP reported an issue in sccache, not assimp.

Symbols are required by all serious developers, you cannot debug field issues without symbols.
In 64bit you can't even walk the stack frames!

Please close this as "By Design".

#5710 reverted this break and added an option to disable symbols if someone really desperately needs to make their binaries impossible to debug.

**Comment by mwestphal** [comment]
> The OP reported an issue in sccache, not assimp.
> 
> Symbols are required by all serious developers, you cannot debug field issues without symbols. In 64bit you can't even walk the stack frames!
> 
> Please close this as "By Design".
> 
> [#5710](https://github.com/assimp/assimp/pull/5710) reverted this break and added an option to disable symbols if someone really desperately needs to make their binaries impossible to debug.

What if you build in release to ship to your users and not to develop anything ?

use proper REL_WITH_DEB_INFO when you need deb info.

**Comment by RichardTea** [comment]
If your tooling does not support PDBs, you can disable `ASSIMP_INSTALL_PDB` as described in #5745 
That disables their creation.

I strongly recommend that you set `/Z7` if do that though, because otherwise if you do not have the PDB from the specific build it will be extremely difficult, if not impossible for you to determine what went wrong when a user sends you a dump file.
In many cases you can't even get a stack trace.

With MSVC, you need the PDB from the specific build you shipped. Rebuilding the same source will _not_ produce a PDB you can use with any other build.

Sadly #4896 created a very serious issue for us. It created _almost empty_ PDBs, which meant it did not break our build and went unnoticed for a few months - until Assimp crashed for one of our users **and we could not determine the cause**.

Kitware have been aware for about five years that their MSVC defaults for Release and MinSizeRel need to set `/Zi`, however there has been a lot of bikeshedding about which other optimisation flags should be default, which has delayed their fix.

In practice, almost everyone uses Release or MinSizeRel and sets `/Zi`.

CMake REL_WITH_DEB_INFO is not appropriate for an MSVC release build as it disables many optimisations, notably inlining small functions. It is really a debug build that links to the release CRT DLLs - this is **very** useful for a developer, but is significantly slower.

**Comment by mwestphal** [comment]
You are absolutely right that being able to set `/Z7` is critical, and having an option that lets you set it is great, I do not really see why it should be the default but having an options is definitely better than having nothing and relying on a non-configurable behavior and patches is definitely not great for anyone.

Being able to debug dump file is indeed a great, being able to not generate .pdb is also great, what matters is configurability :)

> Kitware have been aware for about five years that their MSVC defaults for Release and MinSizeRel need to set /Zi, however there has been a lot of bikeshedding about which other optimisation flags should be default, which has delayed their fix.

You may want to reach out to Kitware: https://www.kitware.com/ and open an issue on CMake gitlab: https://gitlab.kitware.com/cmake/cmake

I've found this issue where you definitely want to raise your concern: https://gitlab.kitware.com/cmake/cmake/-/issues/20812

Anyway, that doesn't concern assimp.

In any case, I don't see why this is still open ? You added an option which seems perfectly fine to use. IMO it should be off by default but just being able to control this behavior is enough.

**Comment by kimkulling** [comment]
Sorry, misleaded the epic.
